pkg/recipes/configloader/environment.go

```diff
@@ -114,6 +114,17 @@ func getConfiguration(environment *v20231001preview.EnvironmentResource, applica
 
 	if environment.Properties.RecipeConfig != nil {
 		config.RecipeConfig = envDatamodel.Properties.RecipeConfig
+		secrets := config.RecipeConfig.Terraform.Authentication.Git.PAT
```

Comment:
Ideally all the recipe type specific operations should be contained within driver, so configloader is not really the right place for this.